calculator/main.py

```diff
@@ -37,14 +46,21 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    if not isinstance(operation, dict):
+        raise InvalidOperation('Given operation is invalid.')
+        
+    else:
+        calc['operations'].update(operation)
 
 
 def get_operations(calc):
     """
     Returns the list of operation names supported by given calculator.
     """
-    pass
+    result = []
```

Comment:
You can accomplish the same thing here with just `return list(calc['operations'])` instead of iterating through and manually constructing the list.